scripts/lib/api/generateToc.ts

```diff
@@ -111,6 +123,64 @@ function addItemsToModules(
   }
 }
 
+function groupAndSortModules(
+  moduleGrouping: TocGrouping,
+  tocModules: TocEntry[],
+  tocModulesByTitle: Dictionary<TocEntry>,
```

Comment:
It's the same data, but organized into a Lodash dictionary through `keyBy(tocModules, (toc) => toc.title)`. The downside of `.find()` is it's O(n) rather than O(1). Our API generation code is already quite slow, and that would make this even slower.

Reply:
Is the array is big enough to make a difference? If so, we could instead keep the `tocModulesByTitle` arg and iterate through its values.

Reply:
The array is n=140 for historical versions. We do this operation a lot, so that does add up. Instead, I opened https://github.com/Qiskit/documentation/pull/1259

Reply:
Ok that works, thanks